simulation/traffic_simulator/test/src/hdmap_utils/test_hdmap_utils.cpp

```diff
@@ -0,0 +1,3152 @@
+  const double distance_threshold = 10.0;
+  bool include_crosswalk = true;
+  size_t search_count = 100;
+  auto lanelets =
+    hdmap_utils.getNearbyLaneletIds(point, distance_threshold, include_crosswalk, search_count);
+
+  EXPECT_TRUE(lanelets.empty());
+}
+
+/**
+ * @note Test basic functionality; test collision point calculations
+ * correctness with ids of a road and a crosswalk that do intersect.
+ */
+TEST(HdMapUtils, getCollisionPointInLaneCoordinate_intersects)
+{
+  auto hdmap_utils = makeHdMapUtilsInstance();
+
+  lanelet::Id id_crosswalk = 34399;
+  lanelet::Id id_road = 34633;
+  auto distance = hdmap_utils.getCollisionPointInLaneCoordinate(id_road, id_crosswalk);
+
+  EXPECT_TRUE(distance.has_value());
+  EXPECT_GT(distance.value(), 0.0);
+}
+
+/**
+ * @note Test basic functionality; test collision point calculations
+ * correctness with ids of a road and a crosswalk that do not intersect.
+ */
+TEST(HdMapUtils, getCollisionPointInLaneCoordinate_disjoint)
+{
+  auto hdmap_utils = makeHdMapUtilsInstance();
+
+  lanelet::Id id_crosswalk = 34399;
+  lanelet::Id id_road = 34579;
+  auto distance = hdmap_utils.getCollisionPointInLaneCoordinate(id_road, id_crosswalk);
+
+  EXPECT_FALSE(distance.has_value());
+}
+
+/**
+ * @note Test function behavior when called with an id of non existing lanelet.
+ */
+TEST(HdMapUtils, getCollisionPointInLaneCoordinate_invalidLanelet)
+{
+  auto hdmap_utils = makeHdMapUtilsInstance();
+
+  lanelet::Id id_crosswalk = 34399;
+  lanelet::Id id_road_invalid = 1000000;
+
+  std::optional<double> distance;
+  EXPECT_THROW(
+    distance = hdmap_utils.getCollisionPointInLaneCoordinate(id_road_invalid, id_crosswalk),
+    std::runtime_error);
+}
+
+/**
+ * @note Test function behavior when called with an id of non existing crosswalk lanelet.
+ */
+TEST(HdMapUtils, getCollisionPointInLaneCoordinate_invalidCrosswalkLanelet)
+{
+  auto hdmap_utils = makeHdMapUtilsInstance();
+
+  lanelet::Id id_crosswalk_invalid = 1000000;
+  lanelet::Id id_road = 34600;
+
+  std::optional<double> distance;
+  EXPECT_THROW(
+    distance = hdmap_utils.getCollisionPointInLaneCoordinate(id_road, id_crosswalk_invalid),
+    std::runtime_error);
+}
+
+/**
+ * @note Test basic functionality; test conversion to lanelet pose correctness with a point
+ * positioned on a given lanelet with the given matching distance
+ * - the goal is to test a regular usecase of correct conversion.
+ */
+TEST(HdMapUtils, toLaneletPose_correct)
+{
+  auto hdmap_utils = makeHdMapUtilsInstance();
+  const lanelet::Id id_road = 34600;
+  const double yaw = M_PI + M_PI_2 / 3.0;
+  const auto pose = makePose(makePoint(3790.0, 73757.0), makeQuaternionFromYaw(yaw));
+
+  const auto lanelet_pose = hdmap_utils.toLaneletPose(pose, false);
+
+  const auto reference_lanelet_pose =
+    traffic_simulator_msgs::build<traffic_simulator_msgs::msg::LaneletPose>()
+      .lanelet_id(id_road)
+      .s(35.0)
+      .offset(0.0)
+      .rpy(geometry_msgs::msg::Vector3());
```

Comment:
Temporary variables used only once should not be defined and should be assigned directly to arguments.

```suggestion
  const auto lanelet_pose = makeHdMapUtilsInstance().toLaneletPose(makePose(makePoint(3790.0, 73757.0), makeQuaternionFromYaw(M_PI + M_PI_2 / 3.0)), false);

  const auto reference_lanelet_pose =
    traffic_simulator_msgs::build<traffic_simulator_msgs::msg::LaneletPose>()
      .lanelet_id(34600)
      .s(35.0)
      .offset(0.0)
      .rpy(geometry_msgs::msg::Vector3());
```

Reply:
Please correct the other sections in the same manner.